### PR TITLE
Modified .gitignore to properly ignore auto_example files

### PR DESCRIPTION
**Pull request by vighneshbirodkar**
`git` was reporting a lot of untracked files since the examples moved. This fixes that behaviour

### CODE REVIEW

**Comment by stefanv** [comment]
This does not look right.  Why are your auto_examples in the root directory?

**Comment by vighneshbirodkar** [comment]
@stefanv They are not. Why do you think so ?

File: .gitignore [hunk 1]
```diff
@@ -18,12 +18,8 @@ dist
 skimage/version.py
 *.swp
 .coverage
-doc/source/auto_examples/*.py
-doc/source/auto_examples/*.txt
-doc/source/auto_examples/notebook
-doc/source/auto_examples/images/plot_*.png
-doc/source/auto_examples/images/thumb
-doc/source/auto_examples/applications/
+auto_examples/
```
Comment (stefanv) [inline .gitignore]: I guess because of this line.  Does this apply recursively?

Comment (vighneshbirodkar) [inline .gitignore]: @stefanv Yes this does apply recursively.

**Comment by vighneshbirodkar** [comment]
@soupault @stefanv You think this should be merged ?

**Comment by soupault** [comment]
I'm not sure. From my point of view,
```
auto_examples/
~doc/source/auto_examples/blank_image.png
```
should be changed to
```
doc/source/auto_examples/**/*
```

**Comment by vighneshbirodkar** [comment]
@soupault Thanks. I changed it

**Comment by soupault** [comment]
:+1: from my side.

**Comment by stefanv** [comment]
What about `doc/source/auto_examples/images/blank_image.png`?

**Comment by soupault** [comment]
@stefanv, `grep` doesn't indicate anything, nor does the only [related commit](https://github.com/scikit-image/scikit-image/commit/4174d5fef00e5f1b0cbf2c99c44954e12dd1b0e4).

I've found that it was used in [`gen_rst.py`](https://github.com/scikit-image/scikit-image/commit/cce3d7bcc5f285b17e2accce5661cdf8c0c84e92#diff-66a73934caf20725e52beca9674a3d9aR282), but that file doesn't exist anymore.

**Comment by vighneshbirodkar** [comment]
@stefanv You think we should explicitly add that file to not be ignored ?

**Comment by stefanv** [comment]
In that case, let's get rid of it.
​

**Comment by soupault** [comment]
LGTM.

**Comment by blink1073** [comment]
Merging, thanks @vighneshbirodkar!

**Pull request merged**